runtimes/mlflow/tests/test_metadata.py

```diff
@@ -0,0 +1,217 @@
+    Parameters,
+)
+from mlserver.settings import ModelSettings
+
+from mlserver_mlflow.metadata import (
+    InputSpec,
+    DefaultInputPrefix,
+    DefaultOutputPrefix,
+    _get_content_type,
+    _get_shape,
+    to_metadata_tensors,
+    to_metadata,
+)
+
+
+@pytest.mark.parametrize(
+    "input_spec, expected",
+    [
+        (
+            TensorSpec(name="foo", shape=(2, 2), type=np.dtype("int32")),
+            ("INT32", NumpyCodec.ContentType),
+        ),
+        (
+            ColSpec(name="foo", type=DataType.string),
+            ("BYTES", StringCodec.ContentType),
+        ),
+        (
+            ColSpec(name="foo", type=DataType.binary),
+            ("BYTES", Base64Codec.ContentType),
+        ),
+    ],
+)
+def test_get_content_type(input_spec: InputSpec, expected: Tuple[str, str]):
+    datatype, content_type = _get_content_type(input_spec)
+    assert (datatype, content_type) == expected
+
+
+@pytest.mark.parametrize(
+    "input_spec, expected",
+    [
+        (
+            TensorSpec(name="foo", shape=(2, 2), type=np.dtype("int32")),
+            [2, 2],
+        ),
+        (
+            ColSpec(name="foo", type=DataType.string),
+            [-1],
+        ),
+    ],
+)
+def test_get_shape(input_spec: InputSpec, expected: List[int]):
+    shape = _get_shape(input_spec)
+    assert shape == expected
+
+
+@pytest.mark.parametrize(
+    "schema, expected",
+    [
+        (
+            Schema(
+                inputs=[TensorSpec(name="foo", shape=(2, 2), type=np.dtype("int32"))]
+            ),
+            [
+                MetadataTensor(
+                    name="foo",
+                    datatype="INT32",
+                    shape=[2, 2],
+                    tags=Tags(content_type=NumpyCodec.ContentType),
+                )
+            ],
+        ),
+        (
+            Schema(inputs=[TensorSpec(shape=(2, 2), type=np.dtype("int32"))]),
+            [
+                MetadataTensor(
+                    name="input-0",
+                    datatype="INT32",
+                    shape=[2, 2],
+                    tags=Tags(content_type=NumpyCodec.ContentType),
+                )
+            ],
+        ),
+        (
+            Schema(
+                inputs=[
+                    ColSpec(type=DataType.string),
+                    ColSpec(type=DataType.integer),
+                ]
+            ),
+            [
+                MetadataTensor(
+                    name="input-0",
+                    datatype="BYTES",
+                    shape=[-1],
+                    tags=Tags(content_type=StringCodec.ContentType),
+                ),
+                MetadataTensor(
+                    name="input-1",
+                    datatype="INT32",
+                    shape=[-1],
+                    tags=Tags(content_type=NumpyCodec.ContentType),
+                ),
+            ],
+        ),
+    ],
+)
+def test_to_metadata_tensors(schema: Schema, expected: List[MetadataTensor]):
```

Comment:
should we have a case for multiple `TensorSpec` as input

Reply:
Sure! I'll add one.